201-site-to-site-vpn-fqdn-bgp/azuredeploy.json

```diff
@@ -0,0 +1,1109 @@
+      }
+    },
+    "sharedKey": {
+      "type": "string",
+      "defaultValue": "_your_Pre-Shared-Secret",
+      "metadata": {
+        "description": "Shared key (PSK) for IPSec tunnels"
+      }
+    },
+    "vm1Name": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('prefix'),'vm1')]",
+      "metadata": {
+        "description": "name of the VM in subnet1 in VNet1"
+      }
+    },
+    "vm2Name": {
+      "type": "string",
+      "defaultValue": "[concat(parameters('prefix'),'vm2')]",
+      "metadata": {
+        "description": "name of the VM in subnet1 in VNet2"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_B1s",
+      "allowedValues": [
+        "Standard_B1ls",
+        "Standard_B1s",
+        "Standard_DS1_v2",
+        "Standard_DS5_v2"
+      ],
+      "metadata": {
+        "description": "Size of the Virtual Machine"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "administrator username of the VMs"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "administrator password of the VMs"
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Premium_LRS"
+      ],
+      "metadata": {
+        "description": "The storage account type for the disks of the VMs"
+      }
+    },
+    "dnsLabelgtw1PubIP1": {
+      "type": "string",
+      "defaultValue": "[toLower( concat( 'gtw1-ip1-', uniqueString(resourceGroup().id) ))]",
+      "metadata": {
+        "description": "dns name of public IP1 of the VPN Gateway1. Must be lowercase. It should match with the regex: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$."
+      }
+    },
+    "dnsLabelgtw1PubIP2": {
+      "type": "string",
+      "defaultValue": "[toLower( concat( 'gtw1-ip2-', uniqueString(resourceGroup().id) ))]",
+      "metadata": {
+        "description": "dns name of public IP2 of the VPN Gateway1. Must be lowercase. It should match with the regex: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$."
+      }
+    },
+    "dnsLabelgtw2PubIP1": {
+      "type": "string",
+      "defaultValue": "[toLower( concat( 'gtw2-ip1-', uniqueString(resourceGroup().id) ))]",
+      "metadata": {
+        "description": "dns name of public IP1 of the VPN Gateway2. Must be lowercase. It should match with the regex: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$."
+      }
+    },
+    "dnsLabelgtw2PubIP2": {
+      "type": "string",
+      "defaultValue": "[toLower( concat( 'gtw2-ip2-', uniqueString(resourceGroup().id) ))]",
+      "metadata": {
+        "description": "dns name of public IP2 of the VPN Gateway2. Must be lowercase. It should match with the regex: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$."
+      }
+    },
+    "imagePublisher": {
```

Comment:
set some allowed values on the publisher, offer, sku params

Reply:
added the variable:
"imageReference": {
"publisher": "Canonical",
"offer": "0001-com-ubuntu-server-focal",
"sku": "20_04-lts",
"version": "latest"
},